python/foglamp/common/configuration_manager.py

```diff
@@ -209,7 +209,18 @@ def get_entry_val(k):
             category_info.append((row['key'], row['description']))
         return category_info
 
-    async def _read_all_groups(self, root):
+    async def _read_all_groups(self, root, children):
+        async def nested_children(child):
+            if not child:
+                return
+            grand_children = await self.get_category_child(child["key"])
```

Comment:
self.get_category_child(child["key"]), where child[''key"] = category_name
and it will return `key,description,children(including nested category info)`